Carbon.FileSystem/Functions/Grant-CNtfsPermission.ps1

```diff
@@ -0,0 +1,135 @@
+function Grant-CNtfsPermission
+{
+    <#
+    .SYNOPSIS
+    Grants permission on folders and files.
+
+    .DESCRIPTION
+    The `Grant-CNtfsPermission` functions grants permissions to folders and files. Pass the folder/file path to the
+    `Path` parameter, the user/group name to the `Identity` parameter, and the permissions to the `Permission`
+    parameter. By default, the permissions are applied to the folder and inherited to all its subfolders and files. To
+    control how the permissions are applied, use the `ApplyTo` parameter. If you want permissions to only apply to child
+    files and folders, use the `OnlyApplyToChildFilesAndFolders` switch.
+
+    By default, an "Allow" permission is granted. To add a "Deny" permission, set the value of the `Type` parameter to
+    `Deny`.
+
+    All existing, non-inherited permissions for the given identity are removed first. If you want to preserve a
+    user/group's existing permissions, use the `Append` switch.

```

Comment:
Seems backwards to me. `Grant` to me implies append.

I'd opt for `Set` as the verb if the function behaves in this way.

Reply:
This function cares about two parts of an NTFS access rule: an identity and the permission . The underlying object, an access rule, actually has four parts: the permission, its type (allow/deny), the identity, and how to apply the permission to ancestors. When our automation grants permissions, 99% of the time, that's all we care about: a specific identity gets/nee. I don't even think the ability to append extra access rules came from us. It might have been a PR.

Your comment got me thinking, and maybe what we should also have are `Add-CNtfsAccessRule`, `Remove-CNtfsAccessRule`, and `Set-CNtfsAccessRule` functions that manage the actual access rules and allow more advanced scenarios. I'll keep thinking about this.